scripts/bwctest_osd.sh

```diff
@@ -264,8 +284,9 @@ function execute_mismatch_tests() {
  
 # setup the cypress test env
 [ ! -d "$TEST_DIR/cypress" ] && setup_cypress
-execute_tests
-(( ${#RELEASES_ARRAY[@]} )) && execute_mismatch_tests
-echo "Completed BWC tests for $TEST_VERSIONS [$DASHBOARDS_TYPE]"
-echo "Total test failures: $TOTAL_TEST_FAILURES"
+[ $GENERATE_DATA == "false" ] && execute_tests || generate_test_data
+[ $GENERATE_DATA == "false" ] && (( ${#RELEASES_ARRAY[@]} )) && execute_mismatch_tests
+[ $GENERATE_DATA == "false" ] &&  echo "Completed BWC tests for $TEST_VERSIONS [$DASHBOARDS_TYPE]"
+[ $GENERATE_DATA == "false" ] &&  echo "Total test failures: $TOTAL_TEST_FAILURES"
+[ $GENERATE_DATA == "true" ] && echo "Generate data complete"
```

Comment:
Would it be cleaner to just check `$GENERATE_DATA` first, and then return/exit early?

Reply:
+1 I am confused on this part too